src/foam/nanos/auth/CompoundAndAuthorizer.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.mlang.predicate.Predicate;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+public class CompoundAndAuthorizer implements Authorizer {
```

Comment:
When would this be useful? Wouldn't we normally just use an Or Authorizer? 

More consistent names would just be AndAuthorizer and OrAuthorizer, as And/Or are already compound operations.